src/include/miopen/interpolate/problem_description.hpp

```diff
@@ -0,0 +1,285 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include "miopen/miopen.h"
+#include <miopen/problem_description_base.hpp>
+#include <miopen/activ.hpp>
+#include <miopen/tensor.hpp>
+#include <cassert>
+#include <string>
+
+namespace miopen {
+
+struct NetworkConfig;
+
+namespace interpolate {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const TensorDescriptor& scaleFactorsDesc_,
+                       const miopenInterpolateMode_t mode_,
+                       const bool align_corners_,
+                       bool is_fwd_)
+        : scaleFactorsDesc(scaleFactorsDesc_),
+          mode(mode_),
+          align_corners(align_corners_),
+          is_fwd(is_fwd_)
+    {
+        IsValidMode();
+    }
+
+    const TensorDescriptor& GetScaleFactorsDesc() const { return scaleFactorsDesc; }
+    miopenInterpolateMode_t GetMode() const { return mode; }
+    bool GetAlignCorners() const { return align_corners; }
+
+    bool IsValidMode() const
+    {
+        if(mode != MIOPEN_INTERPOLATE_MODE_NEAREST && mode != MIOPEN_INTERPOLATE_MODE_LINEAR &&
+           mode != MIOPEN_INTERPOLATE_MODE_BILINEAR && mode != MIOPEN_INTERPOLATE_MODE_TRILINEAR &&
+           mode != MIOPEN_INTERPOLATE_MODE_BICUBIC)
+        {
+            std::cout << "MODE: " << mode << std::endl;
+            MIOPEN_THROW(miopenStatusBadParm, "Interpolate: Invalid mode.");
+        }
+        return true;
+    }
+
+    bool IsValidStride(TensorDescriptor td) const
```

Comment:
not sure why you need this function and the real purpose of this function. But if your purpose is to check stride of a tensor is valid and I read logic of the function, it will definitely return false if you pass an unpacked tensor into it. I think this logic only check if tensor is unpacked or not. And because you used this function in ProblemDescription constructor, user cannot use the API with unpacked tensor. Example for you: lengths={3,2,2}, strides={8,2,1}. I think you don't identify this bug because in your driver, the C flag (to specify input tensor is cont or not) use the tranpose method to convert cont tensor to uncont tensor, however the tensor is still a packed tensor. You can reference to method I use in [this PR](https://github.com/moreh-dev/MIOpen/pull/40/files#) to convert tensor to unpacked tensor (also uncont tensor)

Reply:
removed the function in commit [27b8624](https://github.com/moreh-dev/MIOpen/pull/41/commits/27b8624c97d4aa8cb470b0dfae4ffe9e9cbfebfd)

Reply:
ok